### PR TITLE
Update to lucene snapshot e7c625430ed

### PR DESCRIPTION
**Pull request by romseygeek**
Includes LUCENE-9148 and LUCENE-9398, which splits the BKD metadata, index and data into separate files and keeps the index off-heap.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java [hunk 1]
```diff
@@ -774,14 +776,16 @@ public void testIndexHasNoDuplicateData() throws IOException {
...
             }
-            try (IndexInput in = dir.openInput("bkd", IOContext.DEFAULT)) {
-                in.seek(indexFP);
-                BKDReader r = new BKDReader(in);
-                assertFalse(pointsHaveDuplicateData(r, r.getDocCount()/2));
+            try (IndexInput metain = dir.openInput("bkdmeta", IOContext.DEFAULT);
+                 IndexInput indexin = dir.openInput("bkdindex", IOContext.DEFAULT);
+                 IndexInput datain = dir.openInput("bkddata", IOContext.DEFAULT)) {
+                BKDReader r = new BKDReader(metain, indexin, datain);
+                assertTrue(pointsHaveDuplicateData(r, r.getDocCount() / 2));
```
Comment (romseygeek) [inline server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java:788]: This test is now failing - @iverase could you take a look?

File: server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java [hunk 2]
```diff
@@ -748,14 +748,16 @@ public void testIndexHasDuplicateData() throws IOException {
                 LongPoint.encodeDimension(value, longBytes, 0);
                 w.add(longBytes, docId);
             }
-            long indexFP;
-            try (IndexOutput out = dir.createOutput("bkd", IOContext.DEFAULT)) {
-                indexFP = w.finish(out);
+            try (IndexOutput metaout = dir.createOutput("bkdmeta", IOContext.DEFAULT);
+                 IndexOutput indexout = dir.createOutput("bkdindex", IOContext.DEFAULT);
+                 IndexOutput dataout = dir.createOutput("bkddata", IOContext.DEFAULT)) {
+                w.finish(metaout, indexout, dataout).run();
```
Comment (iverase) [inline server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java:754]: You need to change the finish method to something like:

```
Runnable finalizer = writer.finish(metaOut, indexOut, dataOut);
      if (finalizer != null) {
        metaOut.writeInt(fieldInfo.number);
        finalizer.run();
      }
```

Comment (iverase) [reply server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java:754]: Or even better like this:

```    
        Runnable finalizer = w.finish(out, out, out);
        long indexFP = out.getFilePointer();
        finalizer.run();
 ```

Then when you call the reader you need to make sure retain is in the right position:

```
metain.seek(indexFP);
```

**Review by iverase — APPROVED** [review]
lgtm

**Pull request merged**